src/test/java/com/google/devtools/build/lib/remote/RemoteSpawnCacheTest.java

```diff
@@ -503,4 +503,29 @@ public Void answer(InvocationOnMock invocation) {
         .containsExactly(Pair.of(ProgressStatus.CHECKING_CACHE, "remote-cache"));
     assertThat(eventHandler.getEvents()).isEmpty(); // no warning is printed.
   }
+
+  @Test
+  public void failedCacheActionAsCacheMiss() throws Exception {
+    ActionResult actionResult = ActionResult.newBuilder().setExitCode(1).build();
+    when(remoteCache.getCachedActionResult(any(ActionKey.class)))
+        .thenAnswer(
+            (Answer<ActionResult>) invocation -> {
+              RequestMetadata meta = TracingMetadataUtils.fromCurrentContext();
+              assertThat(meta.getCorrelatedInvocationsId()).isEqualTo("build-req-id");
+              assertThat(meta.getToolInvocationId()).isEqualTo("command-id");
```

Comment:
Why do we need to test for the metadata?

Reply:
Right, we don't. Cleaned up.